### DISCUSSION TITLE
Block applications with changed name

### DISCUSSION THREAD

**Discussion by IwisGiwis** [SafeExamBrowser/seb-win-refactoring#462] [Q&A] [answered]
Dear all,

when you change the name of a .exe file of some applications, SEB does not block the application anymore.  
For example you have AnyDesk.exe in the list of prohibited processes, download AnyDesk and then change AnyDesk.exe into AnyDesk42.exe. Because of that the process name also changes into AnyDesk42.exe and SEB does not block the application.  
</br>
![grafik](https://user-images.githubusercontent.com/74247647/188614360-8e5c5fb8-c755-4009-9f6b-496cfbb78a01.png)

To block the application I have to also block AnyDesk42.exe but I don't think there is a way to block every possible .exe name.
Is there a solution to this problem that I can't find?

Best regards.
Stefan

**Comment by dbuechel** [comment, marked as answer]
Yes, you can (and should) specify the original file name of each application, see https://safeexambrowser.org/windows/win_usermanual_en.html#ProhibitedProcessesSection. The original file name can be determined in the properties dialog of Windows Explorer (right-click the executable -> "Properties" -> "Details").

**Reply by IwisGiwis** [reply]
Thanks for the quick reply. The problem is some .exe files do not have an original file name.  
</br>
![grafik](https://user-images.githubusercontent.com/74247647/188847301-87c5af77-057e-4512-b5a4-5875bc2c60be.png)

**Reply by danschlet** [reply]
Unfortunately there is little we can do about that. I would try to complain to software companies which release applications not having the original file name set. Also we hope that Microsoft at some point realizes that they should provide a secure built-in kiosk/lockdown mode in Window similar to what Apple does in iOS and macOS. Windows for example doesn't even have all its own executables code signed (which could help in detecting non-system executables), which is somehow ridiculous.

**Reply by IwisGiwis** [reply]
I can understand that very well. Anyway you already helped me out with the hint of the original file names. Supremo for example has an original file name and now i can block it.
Thank you very much!

**Reply by dbuechel** [reply]
@danschlet We might be able to improve the process monitoring by also including build version, size, checksum and signature in the application configuration, although that could lead to other issues like configuration overhead (due to version dependence), performance impact (especially in virtual machines) etc.

**Reply by danschlet** [reply]
@dbuechel We probably should investigate if we could include the application signing identity certificate finger print into process monitoring, as that usually should stay the same for updates of the same application/same vendor. But even that could eventually change, so it won't be a perfect solution either.

**Comment by dxdemetriou** [comment]
Why no by Product Name or File Description like:
````wmic process where "name like '%AnyDesk%'" call terminate````
````wmic process where "description like '%AnyDesk%'" call terminate````
?

**Comment by dbuechel** [comment]
We will be extending the application monitoring with a configuration option to verify the signature of an application executable, see #593.
